docs/source/indexing.rst

```diff
@@ -0,0 +1,82 @@
+Indexing Rules
+==============
+
+Data objects can be indexed similarly to NumPy arrays or tensors.
+
+1. **Indexing as tuples**
+
+   - As in NumPy, an index is always a tuple of indices. If it is not, it is converted to a tuple. The elements of the tuple are matched to the dimensions of the data from left to right.
+
+   - Example::
+
+       data[0, 1] == data[(0, 1)]  # The index is converted to a tuple
+       data[(0, 1), ] != data[(0, 1)] 
+	   data[(0, 1), ] == data[((0, 1), )]
+
+2. **Broadcasting**
+
+   - Indexing operations are performed on the fields of a dataclass in the following sequence:
+
+     1. Broadcasting to the dataclass's shape.

```

Comment:
Just to verify, this is broadcasting the shape of the parent dataclass to all children? i.e. if I index kdata along k0 then acqidx is broadcasted first to the shape of kdata and then indexed. But you could not index kdata.header.acq_info along k0 directly, could you?

Reply:
yes.

the shape goes only upwards.the shape of a dataclass is always larger than of its fields.
something to also describe somewhere.

so, an acqidx will have shape k0=1, acq_info will be shape k0=1
but kdata will be the broadcasted shape of all acqinfo, data, traj

I was considering allowing indexing into shape 1 dimensions, so you could index.
so you could index into n_other=1 with (0,1,2),... and you would get n_other=3, indexing with any slice or single integer will keep the shape as n_other=1.
so kdata[0]=kdata[12322] if n_other==1.
(this is how indexing  would be implemented anyways to keep broadcasted dimensions broadcasted, just that there is an explicit check to raise an error if indexed out of bounds.)